### PR TITLE
ADR 007 C4 & PlantUML

### PR DESCRIPTION
**Pull request by psmulovics**

### CODE REVIEW

**Comment by BalassaMarton** [comment]
Couldn't agree more.

File: architecture/adr-007-use-c4.md [hunk 1]
```diff
@@ -10,14 +10,21 @@ Although primarily aimed at software architects and developers, the C4 model pro
...
+Good software architecture diagrams assist with communication (both inside and outside the development and product teams), onboarding new staff, easy of contribution, risk identification, threat modelling, etc. Also, good software architecture diagrams help to align everybody's understanding of the software being built, helping therefore the team to be more efficient.
+
+When it comes to actually to render C4, once you have the architecture models composed of the common SDL format, there are a number of tools available to render them, like PlantUML, Mermaid, Structurizr, and more.
 
 ## Decision
 
 Within ComposeUI, we are to position C4 as the preferred way to capture system architecture and how it is deployed, both as a a means to creating consistent and easily to understand diagrams as code that can live in the source control, and also as a data model that can be used to feed static analysis systems thereby reducing the overhead for developers and improving accuracy of the decisions and suggestions of those systems.
 
+To describe the actual C4, we would use PlantUML extensions and rendering.
```
Comment (BalassaMarton) [inline architecture/adr-007-use-c4.md:21]: Why PlantUML?

Comment (psmulovics) [reply architecture/adr-007-use-c4.md:21]: Sure, what else? You prefer Mermaid?

**Review by raksanyi — COMMENTED** [review]
LGTM

File: architecture/adr-007-use-c4.md (hunk 1 above)
Comment (BalassaMarton) [reply architecture/adr-007-use-c4.md:21]: Scratch that, now that I installed it for eval, I can see that no server is needed.

Comment (psmulovics) [reply architecture/adr-007-use-c4.md:21]: Indeed, wanted to explain in details that you do not need a server - you can just store the DSL in your source control and when docs are built to http://opensource.morganstanley.com/ComposeUI/, that would convert them (and also VSCode plugins do that as you are editing; we can have a compromise solution as well that we do check in the generated png along with the DSL so the pictures would be visible also when you are browsing the github site).

**Comment by psmulovics** [comment]
Can someone from @morganstanley/compose-developers provide a final review before merge?

File: architecture/adr-007-use-c4.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+Good software architecture diagrams assist with communication (both inside and outside the development and product teams), onboarding new staff, easy of contribution, risk identification, threat modelling, etc. Also, good software architecture diagrams help to align everybody's understanding of the software being built, helping therefore the team to be more efficient.
+
+When it comes to actually to render C4, once you have the architecture models composed of the common SDL format, there are a number of tools available to render them, like PlantUML, Mermaid, Structurizr, and more.
+
+## Decision
+
+Within ComposeUI, we are to position C4 as the preferred way to capture system architecture and how it is deployed, both as a a means to creating consistent and easily to understand diagrams as code that can live in the source control, and also as a data model that can be used to feed static analysis systems thereby reducing the overhead for developers and improving accuracy of the decisions and suggestions of those systems.
```
Comment (Sly1024) [inline architecture/adr-007-use-c4.md:19]: Just typos: "a a means" => "a means" and "easily to understand" => "easy to understand"

**Review by Sly1024 — CHANGES_REQUESTED** [review]

**Review by Sly1024 — APPROVED** [review]
LGTM

**Pull request merged**